llama/speculative.cpp

```diff
@@ -0,0 +1,300 @@
+/**
+ * llama.cpp - commit ba1cb19cdd0d92e012e0f6e009e0620f854b6afd - do not edit this file
```

Comment:
Does this have to manually copied over and be kept up to date with llama.cpp?

Reply:
This is done when following the recently changed vendoring process, see make target `sync`, https://github.com/ollama/ollama/blob/4c066179b0f8604fb1ce1c95e45cdd0332bee240/make/Makefile.sync#L192 and https://github.com/ollama/ollama/blob/main/llama/README.md#vendoring . 

While taking a second look I noticed that I should keep vendoring regarding speculative decoding separated from the llava vendoring stuff. FIXED: https://github.com/ollama/ollama/pull/8134/commits/d539e92eb66cc86047de5bd3e1906349cf9083c4